### PR TITLE
Fix VideoPlayer accessibility issues

### PR DESCRIPTION
**Pull request by joshfarrant**
## Summary

Fixes two Sev-2 `VideoPlayer` a11y issues.

The styling of the controls bar is heavier than it was before, but it's necessary to achieve the required contrast.

## List of notable changes:

- Improved contrast of `VideoPlayer` focus styles by making the focus outline thicker and offsetting it.
- Improved contrast of controls and title.
- To offset the heavier title bar styles, the title bar now hides while the video is playing. This should probably have been default behaviour from day 1.
- Similarly, the controls bar now hides when the cursor or focus leaves the `VideoPlayer`, or after a few seconds of inactivity, and reappears when the cursor or focus returns.
- Made the video tooltip more efficient by being more judicious about when the event listener is added.
- 

## Steps to test:

1. Focus the `VideoPlayer` with the keyboard 
1. Observe the focus styles

## Supporting resources (related issues, external links, etc):

- Closes https://github.com/github/primer/issues/3773
- Closes https://github.com/github/primer/issues/3774

## Contributor checklist:

- [x] All new and existing CI checks pass
- [x] Tests prove that the feature works and covers both happy and unhappy paths
- [x] Any drop in coverage, breaking changes or regressions have been documented above
- [x] New visual snapshots have been generated / updated for any UI changes
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

image

 </td>
<td valign="top">

image

</td>
</tr>

<tr>
<td valign="top">

image

 </td>
<td valign="top">

image

</td>
</tr>
</table>

### CODE REVIEW

File: packages/react/src/VideoPlayer/VideoPlayer.visual.spec.ts-snapshots/Visual-Comparison-VideoPlayer-VideoPlayer-Custom-Play-Icon-1-linux.png
Comment (danielguillan) [inline packages/react/src/VideoPlayer/VideoPlayer.visual.spec.ts-snapshots/Visual-Comparison-VideoPlayer-VideoPlayer-Custom-Play-Icon-1-linux.png:1]: Is this change intentional? On the [custom play button story](https://primer-5d4bf586e1-26139705.drafts.github.io/brand/storybook/?path=/story/components-videoplayer-features--custom-play-icon) there's now no contrast between the play icon and the background.

Comment (joshfarrant) [reply packages/react/src/VideoPlayer/VideoPlayer.visual.spec.ts-snapshots/Visual-Comparison-VideoPlayer-VideoPlayer-Custom-Play-Icon-1-linux.png:1]: Good spot, thanks Dani 🙌

I've just updated the colour of the icon for this example

File: packages/design-tokens/src/tokens/functional/components/video-player/colors.js [hunk 1]
```diff
@@ -27,8 +27,8 @@ module.exports = {
       },
       controls: {
         bgColor: {
-          value: 'linear-gradient(var(--base-color-scale-transparent), #01040966)',
-          dark: 'linear-gradient(var(--base-color-scale-transparent), #01040966)',
+          value: 'rgba(0,0,0,0.75)',
+          dark: 'rgba(0,0,0,0.75)',
```
Comment (rezrah) [inline packages/design-tokens/src/tokens/functional/components/video-player/colors.js:31]: Can these be changed to hex format please. That's the preferred format for color values in Primitives now.

```suggestion
          value: '#000000bf',
          dark: '#000000bf',
```

File: packages/react/src/VideoPlayer/VideoPlayer.module.css [hunk 2]
```diff
@@ -48,20 +47,21 @@
...
 }
 
-.VideoPlayer__playButtonOverlay {
+.VideoPlayer__playButtonOverlay.VideoPlayer__playButtonOverlay--transparent {
+  background: transparent;
+}
+
+.VideoPlayer__playButtonOverlay:focus-visible {
+  outline: var(--brand-borderWidth-thicker) solid var(--brand-color-focus);
+  outline-offset: 2px;
```
Comment (rezrah) [inline packages/react/src/VideoPlayer/VideoPlayer.module.css:56]: We have a 2px base token now 😌 

```suggestion
  outline-offset: var(--base-size-2);
```

File: packages/react/src/VideoPlayer/VideoPlayer.module.css [hunk 3]
```diff
@@ -228,9 +233,8 @@
 }
 
 .VideoPlayer__rangeInput:focus-visible {
-  border-color: var(--brand-color-focus);
-  outline: none;
-  box-shadow: 0 0 0 0.125rem var(--brand-color-focus);
+  outline: var(--brand-borderWidth-thick) solid var(--brand-color-focus);
+  outline-offset: 4px;
```
Comment (rezrah) [inline packages/react/src/VideoPlayer/VideoPlayer.module.css:237]:
```suggestion
  outline-offset: var(--base-size-4);
```

File: packages/react/src/VideoPlayer/VideoPlayer.stories.module.css [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.customPlayIcon {
+  color: white;
```
Comment (rezrah) [inline packages/react/src/VideoPlayer/VideoPlayer.stories.module.css:2]: I know it's just a story, but to avoid triggering any linters in future...

```suggestion
  color: var(--base-color-scale-white-0);
```

File: packages/react/src/VideoPlayer/VideoPlayer.visual.spec.ts-snapshots/Visual-Comparison-VideoPlayer-VideoPlayer-With-Poster-1-linux.png
Comment (rezrah) [inline packages/react/src/VideoPlayer/VideoPlayer.visual.spec.ts-snapshots/Visual-Comparison-VideoPlayer-VideoPlayer-With-Poster-1-linux.png:1]: The permanent bg overlay is both quite stark and obstructs some of the video. If we're going this far, can we just position the control bar underneath the video instead of on top? Right now, it's eating into a lot of the video, and the opacity is barely going to be noticed (couldn't see it on my external display)
